### PR TITLE
OTA-1292: upgrade-status: handle unschedulable nodes better

### PR DESCRIPTION
**Pull request by hongkailiu**
https://issues.redhat.com/browse/OTA-1292

Addressing the 2nd case from https://github.com/openshift/oc/pull/1838#pullrequestreview-2238631410

/cc @petr-muller @Davoska

### CODE REVIEW

File: pkg/cli/admin/upgrade/status/mco/layered_node_state.go [hunk 1]
```diff
@@ -207,7 +207,9 @@ func isNodeDoneAt(l *LayeredNodeState, pool *mcfgv1.MachineConfigPool) bool {
 }
 
 const (
-	ReasonOfUnavailabilityMCDWorkInProgress = "MCDWorkInProgress"
+	// ReasonOfUnavailabilityMCDWorkInProgress indicates MCD will fix the state and no user intervention is required.
+	ReasonOfUnavailabilityMCDWorkInProgress = "Machine Config Daemon is processing the node"
+	ReasonOfUnavailabilityNodeUnschedulable = "Node is marked unschedulable"
```
Comment (Davoska) [inline pkg/cli/admin/upgrade/status/mco/layered_node_state.go:212]: In different similar assignments of the `ReasonOfUnavailability` field, we start the reason with a lowercase letter `node`. I don't mind either lowercase or uppercase first letter, and I do acknowledge that we don't control everything, as we propagate some information without modification. However, my goal is to try to be as consistent as possible with our messages :raised_hands: 

`Machine Config Daemon is processing the node` is fine as MCD is a name. English is not my first language, but we do refer to Machine Config Daemon with capitalized letters in [documentation](https://docs.openshift.com/container-platform/4.9/nodes/nodes/nodes-nodes-machine-config-daemon-metrics.html).

```suggestion
ReasonOfUnavailabilityNodeUnschedulable = "node is marked unschedulable"
```

**Review by Davoska — COMMENTED** [review]
Everything is great :raised_hands:

I have only one nitpick regarding letter cases :nerd_face:

File: pkg/cli/admin/upgrade/status/mco/layered_node_state.go (hunk 1 above)
Comment (petr-muller) [reply pkg/cli/admin/upgrade/status/mco/layered_node_state.go:212]: Let's use uppercase please. I believe Hongkai addresses my [comment](https://github.com/openshift/oc/pull/1838#discussion_r1717210542) here. We used uppercase before #1838.

**Review by petr-muller — APPROVED** [review]

File: pkg/cli/admin/upgrade/status/mco/layered_node_state.go (hunk 1 above)
Comment (Davoska) [reply pkg/cli/admin/upgrade/status/mco/layered_node_state.go:212]: Discussion resolved :100:

Comment (Davoska) [reply pkg/cli/admin/upgrade/status/mco/layered_node_state.go:212]: Although, in the future, we should update the other reasons of unavailibility to be consistent.

Comment (petr-muller) [reply pkg/cli/admin/upgrade/status/mco/layered_node_state.go:212]: Yeah, even to the point to uppercasing the first letter of the strings that we take from other sources, such as `failed to drain node`

**Pull request merged**